modules/gapi/include/opencv2/gapi/gcommon.hpp

```diff
@@ -263,12 +265,33 @@ struct graph_dump_path
 };
 /** @} */
 
+/**
+ * @brief Ask G-API to use threaded executor when cv::GComputation
+ * is compiled via cv::GComputation::compile method.
+ *
+ * Specifies a number of threads that should be used by executor.
+ */
+struct use_threaded_executor
+{
+    explicit use_threaded_executor(
+            uint32_t nthreads = static_cast<uint32_t>(cv::getNumThreads()))
```

Comment:
probably it's not ok to have `cv::getNumThreads()` called in the header.

Instead, you could introduce two constructors, a default one `()` with `nthreads` set to `cv::getNumThreads()` in its initializer list, and the value-based one.

Reply:
Can't we just declare it here and move implementation to a cpp file?

Reply:
> Can't we just declare it here and move implementation to a cpp file?
In this case it's needed to call `cv::getNumThreads()` in header anyway